### ISSUE TITLE
Boundary Conditions Violation

### ISSUE THREAD

**Issue by PouyaTaraghi** [lululxvi/deepxde#505]
Hi,
I am trying to solve a ODE of a Euler Beam with these boundary conditions:
u(x=0) = 0
du/dx(x=0)=0
u(x=1)=0
du/dx(x=1)=0
But the predicted results are not satisfying. Actually, the prediction model never consider the boundary conditions of (u(x=0)=0, and u(x=1)=0). Also, there is a big difference between the obtained results from several runs! As you see in the plots, the value of u is not equal to zero (as specified in boundary conditions). I would appreciate it if you let me know the problem.

import deepxde as dde
import numpy as np

geom = dde.geometry.Interval(0,1)

def u_xx(x, u):
    return dde.grad.hessian(u, x)

def u_xxx(x, u):
    return dde.grad.jacobian(u_xx(x, u), x)

def u_xxxx(x, u):
  return dde.grad.jacobian(u_xxx(x, u), x)

def ode(x, u):  
    return u_xxxx(x, u) + 1

def boundary_start(x, on_boundary):
    return on_boundary and np.isclose(x[0], 0)

def boundary_end(x, on_boundary):
    return on_boundary and np.isclose(x[0], 1)

def func(x):
    return -(1)*(x ** 4) / (24) + (1)*(x ** 3) / (12) - (1)*(x ** 2) /(24)

bc1 = dde.DirichletBC(geom, lambda x: 0, boundary_start)
bc2 = dde.NeumannBC(geom, lambda x: 0, boundary_start)
bc3 = dde.DirichletBC(geom, lambda x: 0, boundary_end)
bc4 = dde.NeumannBC(geom, lambda x: 0, boundary_end)

data = dde.data.PDE(
    geom,
    ode,
    [bc1, bc2, bc3, bc4],
    num_domain=10,
    num_boundary=2,
    solution=func,
    num_test=100,
)
layer_size = [1] + [20] * 3 + [1]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.maps.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001, metrics=["l2 relative error"])
losshistory, train_state = model.train(epochs=10000)

dde.saveplot(losshistory, train_state, issave=True, isplot=True)
![xxx](https://user-images.githubusercontent.com/98785571/151902341-bcabac28-66e0-4b00-b889-8b0caf14c0f9.png)

**Comment by lululxvi** [comment]
Check https://deepxde.readthedocs.io/en/latest/demos/eulerbeam.html

**Comment by PouyaTaraghi** [comment]
> Check https://deepxde.readthedocs.io/en/latest/demos/eulerbeam.html
Dear Dr. Lu, 
Thank you for this. Actually, I followed the exact procedure of the provided link but I faced such problem. I would be grateful if you  let me know what can cause this. Thank You!

**Comment by lululxvi** [comment]
If your code is correct, then check FAQ "Q: I failed to train the network or get the right solution, e.g., large training loss, unbalanced losses.".
